Remove torch/legacy

cpuhrsch opened the pull request:
Summary: Largely unused and hinders current development

Differential Revision: D9925094

soumith commented:
you need to remove unit tests as well, in `test/test_legacy_nn.py`

cpuhrsch commented:
Legacy was used to write the following tests:
1) "TestNN.test_bilinear"
2) "_test_rosenbrock" within "TestOptim" for:
- "test_sgd"
- "test_adam"
- "test_adadelta"
- "test_adagrad"
- "test_adamax"
- "test_rmsprop"
- "test_asgd"
- "test_rprop"
- "test_lbfgs"

Will create issues for new implementations.

"TestLuaReader" was deleted (doesn't need a new implementation).

ezyang reviewed (approved):
+1 −13,137 

QWALITY.

cpuhrsch committed:
Remove torch/legacy (#11823)

Summary:
Largely unused and hinders current development
Pull Request resolved: https://github.com/pytorch/pytorch/pull/11823

Differential Revision: D9925094

Pulled By: cpuhrsch

fbshipit-source-id: 1f53f8f21da85c16ae32e75fe6f796718e39fc92
